### PR TITLE
user-following-part2-fix-bug

### PR DESCRIPTION
**Pull request by phamhuycuong**

### CODE REVIEW

File: app/views/words/_words.html.erb [hunk 1]
```diff
@@ -1,20 +0,0 @@
...
-        <tr>
-          <td><%= word.id %></td>
-          <td><%= word.content.capitalize %></td>
-        </tr>
-      <% end %>
-      <tr><td colspan="2"></td></tr>
-    </table>
-    <%= will_paginate words, renderer: BootstrapPagination::Rails %>
-  </div>
-</div>
```
Comment (phamkykhoi) [inline app/views/words/_words.html.erb]: Checkout this file

File: app/controllers/lessons_controller.rb [hunk 2]
```diff
@@ -37,6 +37,17 @@ def update
     end
   end
 
+  def destroy
+    @lesson = Lesson.find params[:id]
+    @user = @lesson.user
+    if @lesson.destroy
+      flash[:success] = I18n.t "lesson.destroy_success"
+    else
+      flash[:success] = I18n.t "lesson.destroy_error"
```
Comment (phamkykhoi) [inline app/controllers/lessons_controller.rb]: + if @lesson.destroy
+ flash[:success] = I18n.t "lesson.destroy_success"
+ else
+ flash[:success] = I18n.t "lesson.destroy_error"
+ end
Viết thành 
 @lesson.destroy ? flash[:success] = I18n.t "lesson.destroy_success" :  flash[:success] = I18n.t "lesson.destroy_error"

File: app/views/layouts/_header.html.erb [hunk 3]
```diff
@@ -14,19 +13,21 @@
...
           <% end %>
           <li class="dropdown">
             <a href="#" class="dropdown-toggle" data-toggle="dropdown">
               <%= current_user.name.capitalize %> <b class="caret"></b>
             </a>
             <ul class="dropdown-menu">
-              <li><%= link_to I18n.t(:menu_profile), "#" %></li>
-              <li><%= link_to I18n.t(:menu_settings), "#" %></li>
+              <li><%= link_to I18n.t(:menu_profile), current_user %></li>
+              <li><%= link_to I18n.t(:menu_settings), edit_user_path(current_user)  %></li>
```
Comment (khanhkhanh) [inline app/views/layouts/_header.html.erb]: 2 spaces after (current_user)

File: app/views/words/_list_words_head.html.erb [hunk 4]
```diff
@@ -7,12 +7,15 @@
...
-    <%= select_tag "sort_by", options_for_select(Settings.word.sort_by_list, params[:sort_by]) %>
+    <% sort_by_list = [[(I18n.t "word.id"), Settings.word.id], [(I18n.t "word.alphabet"), Settings.word.alphabet]] %>
+    <%= select_tag "sort_by", options_for_select(sort_by_list, params[:sort_by]) %>
   </div>
-  <% Settings.word.radio_button_array_show.each_with_index do |radio_button, index| %>
+  <% radio_button_array_show = [I18n.t("word.passed"), I18n.t("word.learned"), I18n.t("word.unlearned"), I18n.t("word.all")] %>
+  <% radio_button_array_var = [Settings.word.passed, Settings.word.learned, Settings.word.unlearned, Settings.word.all] %>
+  <% radio_button_array_show.each_with_index do |radio_button, index| %>
     <div class="col-md-1" align="center">
       <%= radio_button %>
```
Comment (khanhkhanh) [inline app/views/words/_list_words_head.html.erb]: Remove this line if not use

File: Gemfile [hunk 5]
```diff
@@ -1,5 +1,7 @@
 source "https://rubygems.org"
 
+gem 'friendly_id', '~> 5.0.0'
```
Comment (phamkykhoi) [inline Gemfile]: ' ' => " "

File: Gemfile [hunk 6]
```diff
@@ -1,5 +1,7 @@
...
+gem 'faker'
```
Comment (phamkykhoi) [inline Gemfile]: ' ' => " "

File: app/views/words/_list_words_head.html.erb [hunk 7]
```diff
@@ -7,12 +7,15 @@
...
-      <%= radio_button_tag("learn", Settings.word.radio_button_array_var[index], checked = (params[:learn] == Settings.word.radio_button_array_var[index]) || (params[:learn].blank?)) %>
+      <%= radio_button_tag("learn", radio_button_array_var[index], checked = (params[:learn] == radio_button_array_var[index]) || (params[:learn].blank?)) %>
```
Comment (khanhkhanh) [inline app/views/words/_list_words_head.html.erb]: Try removing the outermost ()

File: db/migrate/20150625061244_create_friendly_id_slugs.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    create_table :friendly_id_slugs do |t|
+      t.string   :slug,           :null => false
+      t.integer  :sluggable_id,   :null => false
+      t.string   :sluggable_type, :limit => 50
+      t.string   :scope
+      t.datetime :created_at
+    end
+    add_index :friendly_id_slugs, :sluggable_id
+    add_index :friendly_id_slugs, [:slug, :sluggable_type]
+    add_index :friendly_id_slugs, [:slug, :sluggable_type, :scope], :unique => true
```
Comment (phamkykhoi) [inline db/migrate/20150625061244_create_friendly_id_slugs.rb]: Not use =>
Use :

File: app/views/words/_list_words_head.html.erb (hunk 4 above)
Comment (phamhuycuong) [inline app/views/words/_list_words_head.html.erb]: Show the text above

File: app/models/relationship.rb [hunk 9]
```diff
@@ -0,0 +1,6 @@
+class Relationship < ActiveRecord::Base
+  belongs_to :follower, class_name: "User"
+  belongs_to :followed, class_name: "User"
+  validates :follower_id, presence: true
```
Comment (dieunb) [inline app/models/relationship.rb]: add a white line above this to some group codes:

**Comment by dieunb** [comment]
OK

**Pull request merged**